Code/GraphMol/MolDraw2D/MolDraw2D.cpp

```diff
@@ -2328,6 +2330,70 @@ void MolDraw2D::extractRadicals(const ROMol &mol) {
   }
 }
 
+// ****************************************************************************
+void MolDraw2D::extractLinkNodes(const ROMol &mol) {
+  PRECONDITION(activeMolIdx_ >= 0, "no mol id");
+  PRECONDITION(static_cast<int>(post_shapes_.size()) > activeMolIdx_,
+               "no space");
+  PRECONDITION(static_cast<int>(annotations_.size()) > activeMolIdx_,
+               "no space");
+  if (!mol.hasProp(common_properties::molFileLinkNodes)) {
+    return;
+  }
+
+  bool strict = false;
+  auto linkNodes = MolEnumerator::utils::getMolLinkNodes(mol, strict);
+  for (const auto &node : linkNodes) {
+    const double crossingFrac = 0.333;
+    const double lengthFrac = 0.333;
+    Point2D labelPt{-1000, -1000};
+    Point2D labelPerp{0, 0};
+    for (const auto bAts : node.bondAtoms) {
+      // unlike brackets, we know how these point
+      Point2D startLoc = at_cds_[activeMolIdx_][bAts.first];
+      Point2D endLoc = at_cds_[activeMolIdx_][bAts.second];
+      auto vect = endLoc - startLoc;
+      auto offset = vect * crossingFrac;
+      auto crossingPt = startLoc + offset;
+      Point2D perp{vect.y, -vect.x};
+      perp *= lengthFrac;
+      Point2D p1 = crossingPt + perp / 2.;
+      Point2D p2 = crossingPt - perp / 2.;
+
+      std::vector<std::pair<Point2D, Point2D>> bondSegments;  // not needed here
+      MolDrawShape shp;
+      shp.points =
+          MolDraw2D_detail::getBracketPoints(p1, p2, startLoc, bondSegments);
+      shp.shapeType = MolDrawShapeType::Polyline;
+      post_shapes_[activeMolIdx_].emplace_back(std::move(shp));
+
+      if (p1.x > labelPt.x) {
+        labelPt = p1;
+        labelPerp = crossingPt - startLoc;
```

Comment:
I haven't followed the logic through completely, but it seems to me that either this line could be outside the condition, and line 2376 likewise, or there's a potential for a division by zero error at line 2386.  Or, more likely, I've missed something obvious.

Reply:
there's more or less no way we won't end up inside one of these if statements given that `labelPt` is initialized to `{-1000,-1000}`.